content/documentation/3-stream-developer-guides/2-streams/3-data-flow-stream.md

```diff
@@ -0,0 +1,120 @@
+It is recommended to use `http://`, `maven://` or `docker://` for applications' URIs.
+
+Let's assume you are running Spring Cloud Data Flow, Skipper servers running on your local development environment.
+
+Register the `UsageDetailSender` source application:
+
+```
+dataflow:>app register --name usage-detail-sender --type source --uri file://Users/igopinathan/dev/git/spring-cloud/spring-cloud-dataflow-samples/dataflow-website/stream-developer-guides/streams/usage-detail-sender/target/usage-detail-sender-0.0.1-SNAPSHOT.jar
+```
+
+Register the `UsageCostProcessor` processor application:
+
+```
+dataflow:>app register --name usage-cost-processor --type processor --uri file:///Users/igopinathan/dev/git/spring-cloud/spring-cloud-dataflow-samples/dataflow-website/stream-developer-guides/streams/usage-cost-processor/target/usage-cost-processor-0.0.1-SNAPSHOT.jar
+```
+
+Register the `UsageCostLogger` sink application:
+
+```
+dataflow:>app register --name usage-cost-logger --type sink --uri file://Users/igopinathan/dev/git/spring-cloud/spring-cloud-dataflow-samples/dataflow-website/stream-developer-guides/streams/usage-cost-logger/target/usage-cost-logger-0.0.1-SNAPSHOT.jar
+```
+
+### Create Stream DSL
+
+Create a streaming pipeline that has the Source, Processor and Sink applications as registered above.
+
+```
+dataflow:>stream create usage-cost-logger --definition "usage-detail-sender | usage-cost-processor | usage-cost-logger"
+```
+
+### Validate Stream DSL
+
+Make sure the app coordinates are correct for the stream.
+
+```
+dataflow:>stream validate usage-cost-logger
+╔═════════════════╤══════════════════════════════════════════════════════════════╗
+║   Stream Name   │                      Stream Definition                       ║
+╠═════════════════╪══════════════════════════════════════════════════════════════╣
+║usage-cost-logger│usage-detail-sender | usage-cost-processor | usage-cost-logger║
+╚═════════════════╧══════════════════════════════════════════════════════════════╝
+
+
+usage-cost-logger is a valid stream.
+╔══════════════════════════════╤═════════════════╗
+║           App Name           │Validation Status║
+╠══════════════════════════════╪═════════════════╣
+║source:usage-detail-sender    │valid            ║
+║processor:usage-cost-processor│valid            ║
+║sink:usage-cost-logger        │valid            ║
+╚══════════════════════════════╧═════════════════╝
+
+```
+
+## Deployment
+
+Deploy to local, Cloud Foundry and Kubernetes
+
+### Local
+
+Deploy the stream:
+
+```
+dataflow:> stream deploy usage-cost-logger
+```
+
+Once the stream is deployed on `Local` development environment, you can look the runtime applications via Dashboard's runtime page or using the SCDF Shell command `runtime apps`.
+The runtime applications show information about where each application is running in the local environment and their log files locations.
+
```

Comment:
Not sure if we need to show the log info here.  
But if we do we need show them a screenshot of where to obtain the log file location from runtime apps and then an example bash shell of them catting or tailing the log.